pkg/client/leaderelectionconfig/config.go

```diff
@@ -42,7 +42,8 @@ func BindFlags(l *componentbaseconfig.LeaderElectionConfiguration, fs *pflag.Fla
 		"of a leadership. This is only applicable if leader election is enabled.")
 	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
 		"The type of resource object that is used for locking during "+
-		"leader election. Supported options are `endpoints` (default) and `configmaps`.")
+		"leader election. Supported options are 'endpoints', 'configmaps', "+
```

Comment:
I'm all for updating the documentation to reflect the new options. A little disturbed  by us not having a working default. I believe endpoints should be the default as it was the original behavior and so that is the backward compatible behavior.

Reply:
The default is whatever `l.ResourceLock` string is passed to `BindFlags` in the configuration. In https://github.com/kubernetes/autoscaler/pull/2012 this was intentionally changed from "endpoints" to "leases", which is why the help text was no longer correct.

Any code that still passes "endpoints" in the options to this function will continue to have endpoints as the default.